numba/funcdesc.py

```diff
@@ -131,19 +131,20 @@ def __repr__(self):
         return "<function descriptor %r>" % (self.unique_name)
 
     @classmethod
-    def _get_function_info(cls, interp):
+    def _get_function_info(cls, func_ir):
         """
         Returns
         -------
         qualname, unique_name, modname, doc, args, kws, globals
 
         ``unique_name`` must be a unique name.
         """
-        func = interp.bytecode.func
-        qualname = interp.bytecode.func_qualname
+        func = func_ir.func_id.func
+        qualname = func_ir.func_id.func_qualname
+        # XXX to func_id
```

Comment:
I've experimented with this, and it's not that easy. With `generated_jit`, there is a distinction between the function being implemented (the decorated top-level function) and the implementation being compiled. The `func_id` denotes the implementation being compiled...